services/shuttle-salvo/src/lib.rs

```diff
@@ -27,8 +27,19 @@ impl shuttle_runtime::Service for SalvoService {
     /// Takes the router that is returned by the user in their [shuttle_runtime::main] function
     /// and binds to an address passed in by shuttle.
     async fn bind(mut self, addr: SocketAddr) -> Result<(), Error> {
+        use salvo::prelude::*;
+
+        #[handler]
+        async fn healthz_handler(res: &mut Response) {
+            res.set_status_code(StatusCode::OK);
+        }
+
+        let healthz_router = salvo::Router::with_path("healthz").get(healthz_handler);
+
+        let app = salvo::Router::new().push(healthz_router).push(self.0);
```

Comment:
Will this make the healthz route the innermost route (so middleware might be applied to it)? Is it possible with salvo to just add the healthz route to the router in `self.0`?

Reply:
Unfortunately, `salvo` doesn't provide a way to attach a nested path for an existing router. The only way to nest a route under an existing `Router`  is using `.push()`. [here is an example form the salvo repo](https://github.com/salvo-rs/salvo/blob/42677bfe9fa2dcb7a8f88b378875875b50626f26/examples/todos/src/main.rs#L26)

As for the middlewares, i don't think the middlewares the user defines are going to run on the `healthz_router` since we are pushing it first. Because the `.hoop()` method is how middlewares are added and the docs say middlewares are only applied to the router's descendants [here](https://docs.rs/salvo/latest/salvo/prelude/struct.Router.html#method.hoop)

Reply:
Cool @frectonz ! I think your response resolves O comment.